tests/openedx_learning/apps/authoring/backup_restore/test_restore.py

```diff
@@ -0,0 +1,144 @@
+        # Mock load_dump_zip_file to return our in-memory zip file
+        mock_load_dump_zip_file.return_value = LearningPackageUnzipper().load(self.zip_file)
+
+        out = StringIO()
+        # You can pass any dummy path, since load_dump_zip_file is mocked
+        call_command("lp_load", "dummy.zip", stdout=out)
+
+        lp = self.verify_lp()
+        self.verify_containers(lp)
+        self.verify_components(lp)
+        self.verify_collections(lp)
+
+    def verify_lp(self):
+        """Verify the learning package was restored correctly."""
+        lp = publishing_api.LearningPackage.objects.filter(key=self.lp_key).first()
+        assert lp is not None, "Learning package was not restored."
+        assert lp.title == "Library test"
+        assert lp.description == ""
+        return lp
+
+    def verify_containers(self, lp):
+        """Verify the containers and their versions were restored correctly."""
+        container_qs = publishing_api.get_containers(learning_package_id=lp.id)
+        expected_container_keys = ["unit1-b7eafb", "subsection1-48afa3", "section1-8ca126"]
+
+        for container in container_qs:
+            assert container.key in expected_container_keys
+            draft_version = publishing_api.get_draft_version(container.publishable_entity.id)
+            published_version = publishing_api.get_published_version(container.publishable_entity.id)
+            if container.key == "unit1-b7eafb":
+                assert getattr(container, 'unit', None) is not None
+                assert draft_version is not None
+                assert draft_version.version_num == 1
+                assert published_version is None
+            elif container.key == "subsection1-48afa3":
+                assert getattr(container, 'subsection', None) is not None
+                assert draft_version is not None
+                assert draft_version.version_num == 1
+                assert published_version is None
+            elif container.key == "section1-8ca126":
+                assert getattr(container, 'section', None) is not None
+                assert draft_version is not None
+                assert draft_version.version_num == 1
+                assert published_version is None
+            else:
+                assert False, f"Unexpected container key: {container.key}"
+
+    def verify_components(self, lp):
+        """Verify the components and their versions were restored correctly."""
+        component_qs = components_api.get_components(lp.id)
+        expected_component_keys = [
+            "xblock.v1:drag-and-drop-v2:4d1b2fac-8b30-42fb-872d-6b10ab580b27",
+            "xblock.v1:html:e32d5479-9492-41f6-9222-550a7346bc37",
+            "xblock.v1:openassessment:1ee38208-a585-4455-a27e-4930aa541f53",
+            "xblock.v1:problem:256739e8-c2df-4ced-bd10-8156f6cfa90b",
+            "xblock.v1:survey:6681da3f-b056-4c6e-a8f9-040967907471",
+            "xblock.v1:video:22601ebd-9da8-430b-9778-cfe059a98568",
```

Comment:
Nit: These keys are fine and you don't need to change them. But going forward, you can make more readable keys too. The keys for content that is created in libraries should automatically have a slug in them.